libnmstate/varlink/README.varlink.md

```diff
@@ -0,0 +1,125 @@
+# Varlink support for libnmstate
+Varlink protocol ([Varlink.org](https://varlink.org/)) can used to communicate with libnmstate via stdin/out and varlink client implementation ([Existing Language binding](https://varlink.org/Language-Bindings#how-to-test-new-language-bindings)). Varlink protocol encodes all messages as JSON object and communicates over unix and tcp socket connections. Nmstate service is defined in io.nmstate interface address and libnmstate functions should called in as specified in varlink interface defination file `io.nmstate.varlink`.
+
+## Initiate nmstate varlink service
+unix socket connection
+```
+$ io.nmstate unix:@nmstate &
```

Comment:
I dislike a execute file holding a dot in its name.

Please use `nmstate-varlink` or use `nmstatectl varlink /run/nmstate/io.nmstate`. The second options is used by podman.

Reply:
https://github.com/varlink/varlink.github.io/wiki/Home/f6dc2d249d848a4b18a88b6c2caecec6621968f2#address suggested to use the varlink namespace for the binary:

```
executed binary	exec:/usr/bin/org.example.ftl	Client-only, uses a UNIX socket
```

Reply:
@cathay4t 
What about using namespaced socket like ```@nmstate``` as default. Using optional argument like --address for user to define socket path.

Reply:
@SudhanAruna We are talking about the varlink server binary name. Not the address.

@tyll Even varlink suggest so, I still don't like a varlink service named as `/usr/bin/io.nmstate`.
My preference is let nmstatectl handle it as `nmstate varlink /run/nmstate/nmstate.so`.

The nmstate varlink address __should__ be UNIX socket only where access control is simple.